app/assets/javascripts/comments.coffee

```diff
@@ -29,21 +28,24 @@ new_comment_button_click = (elem) ->
 
     form = $(this).parents('form:first')
 
+    getContent = ()->
+      for editor in tinyMCE.editors
+        if editor.targetElm.form.id == form.attr('id')
+          return editor.getContent()
+
     save = () ->
-        document.start_loading()
-        form.submit()
+      document.start_loading()
+      form.find("textarea").val(getContent())
+      form.submit()
 
     unless document.isLoggedIn
 
-      for editor in tinyMCE.editors
-        if editor.targetElm.form.id == form.attr('id')
-          content = editor.getContent()
-
-      Cookies.set 'new_comment_content', content, { expires: document.expiration_default_time }
+      Cookies.set 'new_comment_content', getContent(), { expires: document.expiration_default_time }
       Cookies.set 'new_comment_is_anonymous', form.find("input[name='comment[is_anonymous]'][type='checkbox']").val(), { expires: document.expiration_default_time }
       Cookies.set 'new_comment_parent_id', form.find('input#comment_parent_id').val(), { expires: document.expiration_default_time }
 
       document.open_login (data) ->
+        document.isLoggedIn = true
```

Comment:
There is a flag named `isLoggedIn` on the document that the file `login.coffee` sets, and is used on this function to verify if the user is logged, but the function `open_login` does not updates this variable, so I had to make this workaround, otherwise, even though the user is logged, the login modal would pop again.